File: frontend/src/types/invoice.ts
export interface InvoiceItem {
  id?: number;
  description: string;
  quantity: number;
  unitPrice: number;
  taxRate: number;
  lineTotal?: number;
}

export interface Invoice {
  id: number;
  invoiceNo: string;
  customerId: number;
  customerName: string;
  branchId: number;
  branchName: string;
  invoiceDate: string;
  dueDate: string;
  subtotal: number;
  discountPercent: number;
  discountAmount: number;
  taxAmount: number;
  grandTotal: number;
  paidAmount: number;
  balanceAmount: number;
  status: string;
  paymentStatus: string;
  approvalStatus: string;
  notes?: string;
  items: InvoiceItem[];
}

export interface Customer {
  id: number;
  code: string;
  name: string;
  email?: string;
  phone?: string;
  gstin?: string;
  addressLine?: string;
  branchId: number;
  branchName: string;
  active: boolean;
}

export interface Payment {
  id: number;
  invoiceId: number;
  invoiceNo: string;
  branchId: number;
  branchName: string;
  paymentDate: string;
  amount: number;
  paymentMode: string;
  referenceNo?: string;
  notes?: string;
  receivedBy: string;
}

File: backend/src/main/java/com/custoking/ims/model/Invoice.java
package com.custoking.ims.model;

import java.util.List;

public record Invoice(
        long id,
        String invoiceNo,
        long customerId,
        String customerName,
        long branchId,
        String branchName,
        String invoiceDate,
        String dueDate,
        double subtotal,
        double discountPercent,
        double discountAmount,
        double taxAmount,
        double grandTotal,
        double paidAmount,
        double balanceAmount,
        String status,
        String paymentStatus,
        String approvalStatus,
        String notes,
        List<InvoiceItem> items
) {}
